src/utils.c

```diff
@@ -235,6 +289,36 @@ app_get_object_info (GApplication *app,
   return FALSE;
 }
 
+const gchar *
+signature_from_type (GType type)
+{
+  if (type == G_TYPE_CHAR || type == G_TYPE_UCHAR)
+    return "y";
+  else if (type == G_TYPE_BOOLEAN)
+    return "b";
+  else if (type == G_TYPE_INT || type == G_TYPE_LONG)
+    return "i";
+  else if (type == G_TYPE_UINT || type == G_TYPE_ULONG)
+    return "u";
+  else if (type == G_TYPE_INT64)
+    return "x";
+  else if (type == G_TYPE_UINT64)
+    return "t";
+  else if (type == G_TYPE_FLOAT || type == G_TYPE_DOUBLE)
+    return "d";
+  else if (type == G_TYPE_VARIANT)
+    return "v";
+  else if (type == G_TYPE_STRV)
+    return "as";
+  else if (type == G_TYPE_STRING || type == G_TYPE_GTYPE || type == G_TYPE_GSTRING ||
```

Comment:
I wonder if these should be in a switch statement (and the rest in ifs), in case that helps readability.

Reply:
well some types are not constants , so you cant really use a switch statement